generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx

```diff
@@ -33,14 +35,91 @@ export interface IUserManagementProps {
   match: any;
 }
 
-export class UserManagement extends React.Component<IUserManagementProps, undefined> {
+export interface IUserManagementState {
+  itemsPerPage: number;
+  items: number;
+  sort: string;
+  order: string;
+  activePage: number;
+  totalItems: number;
+}
+
+export class UserManagement extends React.Component<IUserManagementProps, IUserManagementState> {
 
   constructor(props) {
     super(props);
+    const pageParam = this.getUrlParameter('page', props.location.search);
+    const sortParam = this.getUrlParameter('sort', props.location.search);
+    let sort = 'id';
+    let order = 'asc';
+    let activePage = 1;
+    if (pageParam !== '' && !isNaN(Number(pageParam))) {
+      activePage = Number(pageParam);
+    }
+    if (sortParam !== '') {
+      sort = sortParam.split(',')[0];
+      order = sortParam.split(',')[1];
+    }
+    this.state = {
+      itemsPerPage: ITEMS_PER_PAGE,
+      items: 0,
+      sort,
+      order,
+      activePage,
+      totalItems: 0
+    };
+    this.handlePagination = this.handlePagination.bind(this);
   }
 
   componentDidMount() {
-    this.props.getUsers();
+    this.props.getUsers(this.state.activePage - 1, this.state.itemsPerPage, this.state.sort + ',' + this.state.order).then(action => {
```

Comment:
this is not how it should be handled in redux, you shouldnt handle sideeffect. Just update the store and use the value from store in component